### PR TITLE
[Security] Clean deps

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

This follows the split of security-acl.

### CODE REVIEW

File: src/Symfony/Component/Security/composer.json [hunk 1]
```diff
@@ -37,17 +37,12 @@
...
         "symfony/routing": "~2.2|~3.0.0",
-        "symfony/translation": "~2.0,>=2.0.5|~3.0.0",
         "symfony/validator": "~2.5,>=2.5.5|~3.0.0",
-        "doctrine/common": "~2.2",
-        "doctrine/dbal": "~2.2",
-        "psr/log": "~1.0",
         "symfony/expression-language": "~2.6|~3.0.0",
-        "symfony/ldap": "~2.8|~3.0.0"
+        "symfony/ldap": "~2.8|~3.0.0",
+        "psr/log": "~1.0"
```
Comment (stof) [inline src/Symfony/Component/Security/composer.json]: I suggest keeping psr/log in the same place to make conflicts a bit less likely when merging branches

Comment (nicolas-grekas) [inline src/Symfony/Component/Security/composer.json]: there is already a conflict in both cases (2.7 -> 2.8 and 2.8 -> master) that we have to resolve everytime so this change won't enhance not degrade anything here...

Comment (nicolas-grekas) [inline src/Symfony/Component/Security/composer.json]: oh, and it's already at the last post in master:
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/composer.json#L44

File: src/Symfony/Component/Security/composer.json [hunk 2]
```diff
@@ -37,17 +37,12 @@
         "symfony/finder": "~2.3|~3.0.0",
         "symfony/polyfill-intl-icu": "~1.0",
         "symfony/routing": "~2.2|~3.0.0",
-        "symfony/translation": "~2.0,>=2.0.5|~3.0.0",
-        "symfony/validator": "~2.5,>=2.5.5|~3.0.0",
-        "doctrine/common": "~2.2",
-        "doctrine/dbal": "~2.2",
-        "psr/log": "~1.0",
+        "symfony/validator": "~2.5,>=2.5.9|~3.0.0",
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/composer.json]: 2.5.9 is the version where lowest deps where fixed for symfony/validator

Comment (stof) [inline src/Symfony/Component/Security/composer.json]: shouldn't it be fixed in older branches too then ?

Comment (nicolas-grekas) [inline src/Symfony/Component/Security/composer.json]: true, see #16441

**Comment by nicolas-grekas** [comment]
@stof PR updated

**Comment by stof** [comment]
:+1:

**Pull request merged**